src/engine/QueryPlanner.cpp

```diff
@@ -502,6 +555,15 @@ bool QueryPlanner::checkUsePatternTrick(
                 break;
               }
             }
+          } else if constexpr (std::is_same_v<T, GraphPatternOperation::Bind>) {
+            // check if pattern trick object has been used in Bind
```

Comment:
// If the object variable of ql:has-predicate is used somewhere in a BIND, we cannot use the pattern trick.